### PR TITLE
[icons] feat: "binary-number", "curly-braces"

### PR DESCRIPTION
**Pull request by waynecai1213**
#### Reviewers should focus on:

Add "binary number" and "curly braces" icons.

#### Screenshot

image image

### CODE REVIEW

File: packages/icons/icons.json [hunk 1]
```diff
@@ -4184,5 +4184,20 @@
...
+        "codepoint": 62295
+    },
+    {
+        "displayName": "Curly braces",
+        "iconName": "curly-braces",
+        "tags": "data, type, format，punctuation, grammar",
+        "group": "miscellaneous",
+        "codepoint": 62296
     }
+
```
Comment (adidahiya) [inline packages/icons/icons.json]: this empty line is failing prettier formatting checks

I recommend enabling the Prettier extension for VSCode, that will prevent issues like this from being committed if you format your files on save.

```suggestion
```

File: packages/icons/icons.json [hunk 2]
```diff
@@ -4184,5 +4184,20 @@
         "tags": "footnote, notebook, note, reference, text, citation",
         "group": "editor",
         "codepoint": 62294
+    },
+    {
+        "displayName": "Binary",
+        "iconName": "binary",
```
Comment (adidahiya) [inline packages/icons/icons.json]: I think the names should be more specific here, it's not immediately obvious that we're dealing with numbers. Unlike the `"floating-point"` icon, in this case there is potential ambiguity when dealing with things like binary trees, etc.

```suggestion
        "displayName": "Binary number",
        "iconName": "binary-number",
```

File: packages/icons/icons.json [hunk 3]
```diff
@@ -4184,5 +4184,20 @@
...
+        "tags": "data, type, number, numeric, format",
+        "group": "miscellaneous",
```
Comment (adidahiya) [inline packages/icons/icons.json]: I think this new icon belongs in the "data" group.

```suggestion
        "group": "data",
```

existing icons like `"floating-point"` also should move to the "data" group, but that can happen in a separate PR

File: packages/icons/icons.json [hunk 4]
```diff
@@ -4184,5 +4184,20 @@
...
+        "codepoint": 62295
+    },
+    {
+        "displayName": "Curly braces",
+        "iconName": "curly-braces",
+        "tags": "data, type, format，punctuation, grammar",
```
Comment (adidahiya) [inline packages/icons/icons.json:4198]: can you explain the connection with "data" for this icon?

What is this icon going to be used for?

does it belong in the "editor" group of icons?

**Review by adidahiya — CHANGES_REQUESTED** [review]

File: packages/icons/icons.json (hunk 4 above)
Comment (waynecai1213) [reply packages/icons/icons.json:4198]: We want to use this icon for the "Struct" data type in Pipeline Builder.

File: resources/icons/20px/binary.svg
Comment (adidahiya) [inline resources/icons/20px/binary.svg:1]: you need to update the icon file names to match the updated icon name

Comment (waynecai1213) [reply resources/icons/20px/binary.svg:1]: Updated! thank you!!

**Comment by adidahiya** [comment]
both sizes look good in [docs preview](https://output.circle-artifacts.com/output/job/c0786c51-465c-441b-a8b8-88abb404a297/artifacts/0/packages/docs-app/dist/index.html#icons/icons-list)

image

image

**Review by adidahiya — APPROVED** [review]

**Pull request merged**